modules/core/src/main/java/org/apache/ignite/internal/direct/stream/v2/DirectByteBufferStreamImplV2.java

```diff
@@ -584,7 +588,29 @@ public DirectByteBufferStreamImplV2(MessageFactory msgFactory) {
 
     /** {@inheritDoc} */
     @Override public void writeString(String val) {
-        writeByteArray(val != null ? val.getBytes() : null);
+        if (val != null) {
+            if (buf.capacity() < val.length()) {
+                if (stringsMap == null)
+                    stringsMap = new HashMap<>();
+
+                byte[] bytes = stringsMap.computeIfAbsent(val, s -> s.getBytes());
+
+                try {
+                    writeByteArray(bytes);
+                }
+                catch (Exception e) {
```

Comment:
I believe if an exception appears here, this whole object will be invalid, for example, arrOff would be incorrect, so perhaps it's not necessary to handle this exception here.